src/plugins/auto/auto_schedule.cpp

```diff
@@ -676,23 +678,23 @@ void AutoSchedule::WaitActualNetworkReady() const {
 bool AutoSchedule::ScheduleToWorkerInferRequest(IE::Task inferPipelineTask, DeviceName preferred_device) {
     std::vector<DeviceInformation> devices;
     // AUTO work mode
+    // Devices that fail infer will be removed from the priority list in the callback, need lock here
+    std::unique_lock<std::mutex> lock(_autoSContext->_fallbackMutex);
```

Comment:
here I remove the lock from if else block out, because all the ACTUALDEVICE, FALLBACK or ctput device info may been changed in the fallback logic, please review @WeldonWangwang

Reply:
Okay.